File: 16.cpp
double Power(double base, int exponent){
    if(base == 0)
        return 0;
    if(exponent == 0)
        return 1;
    int unsignedExponent = exponent > 0? exponent: -exponent;
    int ans = 1;
    for(int i = 0; i < unsignedExponent; ++i){
        ans *= base;
    }
    if(exponent > 0)
        return ans;
    else
        return 1.0 / ans;
}

File: 16.java
public class Solution {

    public double Power(double base, int exponent) {
        
       // int flag = base > 0? 1: -1;
        if (exponent == 1)
            return base;
        else if (exponent % 2 == 1)
            return Power(base, exponent - 1) * base;
        else
            return Power(base * base, exponent >> 1);
    }
}
